### ISSUE TITLE
High CPU usage on profile editor

### ISSUE THREAD

**Issue by voed** [Artemis-RGB/Artemis#790]
When the profile editor is open, it fully consumes one CPU core of my Ryzen 5900X. Also import/export dialogs are very laggy. Looks like some dead loop for me. Tried Vulkan renderer, nothing changed.

**Comment by diogotr7** [comment]
What OS is this on? Could you please run the jetbrains profiler (search in the plugins) and post the result?

**Comment by voed** [comment]
Windows 11.

https://drive.google.com/file/d/1T34cxjR0nBc9YqrQFjj9zyTm9jfXMxG6/view?usp=sharing

**Comment by diogotr7** [comment]
Did this start happening with the latest update or was it already present?

**Comment by voed** [comment]
Just installed the app, so can't say. But i have tried the oldest available build ([1.2023.0326.9](https://artemis-rgb.com/releases/windows/0a9856b1-e9b7-4e1f-8f92-b75afa7ba9ee), same result.

**Comment by RobertBeekman** [comment]
Hey, could you tell me the refresh rate of the monitor you are using?
I've not experienced this issue myself on a 165hz monitor but perhaps if you're on something much higher that could explain

**Comment by voed** [comment]
I have 165Hz/2K main and 60Hz/FHD monitors. Tried to force vsync/max framerate for the desktop and Artemis via Nvidia control panel, but nothing changed.

**Comment by camAtGitHub** [comment]
Can confirm 20% CPU usage (1x maxed CPU core and another at about 80% usage) when on keyboard preview / emulator (not sure on exact name)  

Also going to:   
Edit Profile Properties > Activation Conditions  
Then:  
File > Import Script then clicking on / attempting to click different files / browsing for 'Artemis node scripts *.json' files is extremely laggy - Even clicking 'cancel' takes along time to register.

It also crashes with (after clicking cancel or clicking filenames etc)
```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Artemis.UI.Screens.VisualScripting.NodeScriptWindowViewModel.ExecuteImport()
```

Running Windows 10 Pro Version 22H2 / OS Build 19045.2604
Artemis 2 v.1.2023.0518.7

**Comment by RobertBeekman** [comment]
Hi, I did some investigation and the device visualizer is quite heavy on performance.
To alleviate this a little bit I've made it so that it only rerenders if anything changed. This means that when the editor is paused or showing a static effect, CPU usage should drop.

As for the file picker, that was an unrelated bug I've also fixed, thanks for reporting that.

I'll leave this open for now, please let me know if the next update makes things better :)

**Comment by voed** [comment]
Um, looks like the issue is fixed in 1.2023.0522.4. I dont see abnormal cpu usage anymore
